posts/the-radon-nikodym-theorem-in-lean.md

```diff
@@ -0,0 +1,144 @@
+to $\nu$ (denoted $\xi \perp \nu$) and $\mu = \xi + f\nu$.
+As with the Jordan decomposition theorem, it was not clear how to represent this 
+statement. In particular, it is important for us to be able to extract the aforementioned 
+$f$ from the decomposition as this is the Radon-Nikodym derivative. After different
+experimentations, the statement was in the end decided to be represented as 
+a class. 
+```lean
+class have_lebesgue_decomposition (μ ν : measure α) : Prop :=
+(lebesgue_decomposition :
+  ∃ (p : measure α × (α → ℝ≥0∞)), measurable p.2 ∧ p.1 ⊥ₘ ν ∧ μ = p.1 + ν.with_density p.2)
+```
+Since in order to prove the Lebesgue decomposition for $\sigma$-finite measures, 
+we will first need to show it for finite measures, this definition allows us 
+to reuse the same statement for both cases, avoiding duplicate code. Furthermore, 
+as we would like to extract the measure and measurable function from the decomposition, 
+we may define functions that chooses the decomposition if it exists, and 
+zero otherwise.
+```lean
+def singular_part (μ ν : measure α) : measure α :=
+if h : have_lebesgue_decomposition μ ν then (classical.some h.lebesgue_decomposition).1 else 0
+
+def radon_nikodym_deriv (μ ν : measure α) : α → ℝ≥0∞ :=
+if h : have_lebesgue_decomposition μ ν then (classical.some h.lebesgue_decomposition).2 else 0
+```
+With the Lebesgue decomposition for $\sigma$-finite measures formalized, the 
+Radon-Nikodym theorem follows easily. Furthermore, the generalisation of the 
+Radon-Nikodym theorem to signed measures follows also, simply by utilising 
+the Jordan decomposition and realising the Radon-Nikodym derivative of 
+the signed measure is the difference of the Radon-Nikodym derivative of 
+the parts of the Jordan decomposition. However, using this definition of the 
+Radon-Nikodym derivative for the signed measures poses a problem, in which 
+proving any properties about them requires an absolutely continuous condition.
+Thus, it was decided to generalise the Lebesgue decomposition theorem for 
+signed measures from which we obtain the general Radon-Nikodym theorem.
+
+Similar to the Lebesgue decomposition for the positive measures, the Lebesgue 
+decomposition between a signed measure and a positive measure states that, 
+given a signed measure $s$ and a $\sigma$-finite measure $\mu$, there exists 
+a measurable function $f : \alpha \to \mathbb{R}$ and a signed measure $t$, such 
+that $t \perp \mu$ and $s = t + f\mu$.
+While this version of the Lebesgue decomposition was also represented as a class, 
+the statement itself was modified to be an equivalent, yet easier to work with 
+statement. Namely, a signed measure $s$ has Lebesgue decomposition with respect to 
+a measure $\mu$ if both parts of the Jordan decomposition of $s$ have Lebesgue 
+decomposition to $\mu$.
+```lean
+class have_lebesgue_decomposition (s : signed_measure α) (μ : measure α) : Prop :=
```

Comment:
It might make sense to give a more qualified name here (is this `signed_measure.have_lebesgue_decomposition`?) so as to distinguish this from the other `have_lebesgue_decomposition` above.